### DISCUSSION TITLE
How to update within a minor version only?

### DISCUSSION THREAD

**Discussion by l0b0** [renovatebot/renovate#26149] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

[37.87.0 (runner v16.60.1)](https://gitlab.com/engmark/renovate/-/merge_requests/141)

### If you're self-hosting Renovate, select which platform you are using.

gitlab.com

### What is your question?

[This Renovate MR](https://gitlab.com/engmark/vcard/-/merge_requests/275/diffs) breaks my testing strategy, where I want to test my package against several minor versions of Python. I'd still like to get Renovate updates for each minor version, but updating Python 3.8 through 3.11 to 3.12 means the package would no longer be tested adequately.

### Logs (if relevant)

<details><summary>Logs</summary>

```

Copy/paste the relevant log(s) here, between the starting and ending backticks

```

</details>

**Comment by l0b0** [comment]
Is it possible that this is as simple as `separateMinorPatch = true`? Or would this simply end up creating a single PR to upgrade all 3.* dependencies to 3.12.something?

Based on [this MR](https://gitlab.com/engmark/vcard/-/merge_requests/274) it's possible that Renovate is already doing what I want, except for creating an additional MR to upgrade all the minor versions, which I could simply ignore.

**Reply by rarkins** [reply]
separateMinorPatch=true will mean that if both a patch and a minor update exist for the same dependency, then both will be proposed. If you then set a package rule to disable minor updates for that package then it means you'll only get patch updates, when available

**Comment by elchenberg** [comment, marked as answer]
I think you want to disable minor updates with a package rule like this:

```json
  "packageRules": [
    {
      "matchDatasources": [
        "docker"
      ],
      "matchPackageNames": [
        "python"
      ],
      "matchUpdateTypes": ["minor"],
      "enabled": false
    }
  ]
```

Digests will still be updated. If you want to update patch versions, you have to use an image tag with a patch version, for example `3.9.18-alpine` instead of `3.9-alpine`.

**Reply by l0b0** [reply]
Thanks! How would I achieve the same once a new major version comes in? That is, I'd want to test 3.N and 4.0, and to only upgrade their patch/digest versions.

**Reply by elchenberg** [reply]
Adding `major` to the list of `matchUpdateTypes` should do the trick.

**Reply by l0b0** [reply]
Ended up with [this commit](https://gitlab.com/engmark/mypy-exercises/-/commit/eb50e9aa58c5610c563af1b9240afbefa66f7b72). Thank you very much for the help!
